torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -667,20 +668,24 @@ void LoopNest::computeInline(const Buf* b) {
     if (s->buf() == b) {
       auto reductions = NodeFinder<ReduceOp>::find(s);
       if (!reductions.empty()) {
-        throw std::logic_error("cannot inline a reduction computation");
```

Comment:
Why did you change this from throwing to returning false?

Reply:
There are couple of reasons for this:
1. In order to check if a given `buf` corresponds to a reduction, we will have to do the same check in the caller before calling this function. Then, doing the check here seems unnecessary.
2. It is better to have the API `computeInline` **try** to inline the given buf and return if it was successful, rather than crashing the compiler by throwing an exception. (IMO, throwing an exception should only be for the case when the compiler is in a bad state.)
3. The logic to check whether we can inline is best kept in one location, in the function `computeInline` itself. Otherwise, we will have to do these checks in every caller location (assuming this could be called from multiple locations).

Reply:
I'm not 100% convinced, for 2 especially since we throw in other transforms and we'll have to catch above anyway.

Let's get a third opinion from the team.